### DISCUSSION TITLE
Problem whit dimmer light

### DISCUSSION THREAD

**Discussion by lpuntaloro** [HomeSpan/HomeSpan#694] [Q&A]
Hi, I'm having a problem I've never had.. I use homespan with 14 accessories on an esp32. 4 devices are variable lights and have always worked, lately they turn on and off but not variation in brightness. The problem only happens with our 3 iPhones (iOS 17) but it doesn't happen with Watchos or asking Siri.. I also tried with the Xcode simulator and that also makes the same flaw !!!!!! I have already reloaded the firmware but to no avail! Does something like this happen to anyone? Thank you

**Comment by HomeSpan** [comment]
If the Xcode simulator is causing the same behavior, then this is likely an iOS 17 issue.  I noticed that under iOS 17 the brightness bar for a lightbulb behaves differently if I set min/max values.

**Reply by lpuntaloro** [reply]
Hi, I found the cause of the malfunction.
 I initialized some homespan objects (light bulb and dimmer light) by passing an ID parameter in order to perform operations when a specific ID was updated. 
Homekit used to send the parameter but now it no longer sends it so the operating logic no longer carries out its operations.
 The really strange thing is that watchOS works fine instead.....


I used this code to define:

struct DEV_DimmableLED : Service::LightBulb {  // Dimmable LED ESP32 OUT
  int pinOut;
  int ID;

  SpanCharacteristic *power;
  SpanCharacteristic *level;

  DEV_DimmableLED(int pin, int ID)
    : Service::LightBulb() {  // constructor() method
    power = new Characteristic::On();
    level = new Characteristic::Brightness(50);
    level->setRange(1, 100, 1);
    this->pinOut = pin;
    this->ID = ID;
  }  // end constructor

  boolean update() {  // update() method
    bool stato = power->getNewVal();
    if (power->updated()) {  //accendi o spegni
     ...........
    }
 
The problem is here:

>   if (level->updated()) {
>       if (ID == 12) ........
>       if (ID == 13) ........
>     }

    return (true);  // return true
  }                 // update
